incubator/hnc/test/e2e/demo_test.go

```diff
@@ -61,4 +61,23 @@ var _ = Describe("Demo", func() {
 		RunShouldContain(nsTeamB+"-wizard", 5, "kubectl -n", nsService1, "get roles")
 		RunShouldNotContain(nsTeamA+"-wizard", 5, "kubectl -n", nsService1, "get roles")
 	})
+
+	It("Should propagate different types", func(){
+		MustRun("kubectl create ns", nsOrg)
+		MustRun("kubectl hns create", nsTeamA, "-n", nsOrg)
+		MustRun("kubectl hns create", nsTeamB, "-n", nsOrg)
+		MustRun("kubectl create ns", nsService1)
+		MustRun("kubectl hns set", nsService1, "--parent", nsTeamB)
+		MustRun("kubectl -n", nsTeamB, "create secret generic my-creds --from-literal=password=iamteamb")
+		// secret does not show up in service-1 because we haven’t configured HNC to propagate secrets in HNCConfiguration.
+		RunShouldNotContain("my-creds", 5, "kubectl -n", nsService1, "get secrets")
```

Comment:
Because we're waiting for something to _not_ happen, it actually does make sense to do an explicit sleep here (2s should be enough).

Why do you have a 5s timeout here? Was this needed to make the test pass?

Reply:
No I was trying to keep it consistent with the 5s we used in Eventually(). It's basically a random number, and if anything succeeds before 5s, it will return only using the time it need

Reply:
Actually after sleeping 2s, the secrete does get propagated to `service-1`

Reply:
Maybe turn "5" into `testutils.Timeout`?

wdyt "the secret does get propagated?" The code currently says the opposite?

```
		// wait 2 seconds to give time for secret to propogate if it was to
		time.Sleep(2*time.Second)
		// secret does not show up in service-1 because we haven’t configured HNC to propagate secrets in HNCConfiguration.
```

Reply:
Resolved it by setting to ignore the Secret at the beginning of the test